File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/secretsmanager/SecretName.java
package uk.nhs.nhsx.core.aws.secretsmanager;

import uk.nhs.nhsx.core.ValueType;

public class SecretName extends ValueType<SecretName> {
    private SecretName(String value) {
        super(value);
    }

    public static SecretName of(String name) {
        return new SecretName(name);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/secretsmanager/SecretName.kt
package uk.nhs.nhsx.core.aws.secretsmanager

import dev.forkhandles.values.NonEmptyStringValueFactory
import dev.forkhandles.values.StringValue

class SecretName private constructor(value: String) : StringValue(value) {
    companion object : NonEmptyStringValueFactory<SecretName>(::SecretName)
}
